eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/jobs/ClusteringFromMongoJobConfiguration.java

```diff
@@ -35,24 +52,62 @@
+                                      @Qualifier(PROCESS_RS_MERGE_CANDIDATES_STEP) Step processRSMergeCandidatesStep,
+                                      @Qualifier(PROCESS_RS_SPLIT_CANDIDATES_STEP) Step processRSSplitCandidatesStep,
+                                      @Qualifier(CLEAR_RS_MERGE_AND_SPLIT_CANDIDATES_STEP) Step clearRSMergeAndSplitCandidatesStep,
+                                      @Qualifier(CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP) Step clusteringNonClusteredVariantsFromMongoStep,
+                                      // Back-propagate RS that were newly created in the remapped assembly
+                                      @Qualifier(BACK_PROPAGATE_NEW_RS_STEP) Step backPropagateNewRSStep,
+                                      // Back-propagate RS in the remapped assembly that were split or merged
+                                      @Qualifier(BACK_PROPAGATE_SPLIT_MERGED_RS_STEP)
+                                                  Step backPropagateSplitMergedRSStep,
+                                      StepBuilderFactory stepBuilderFactory,
+                                      JobBuilderFactory jobBuilderFactory,
+                                      InputParameters inputParameters) {
+        JobExecutionDecider jobExecutionDecider = isRemappedAssemblyPresent(inputParameters);
+        Step dummyStep = dummyStep(stepBuilderFactory);
         return jobBuilderFactory.get(CLUSTERING_FROM_MONGO_JOB)
-                                .incrementer(new RunIdIncrementer())
-                                .start(clusteringClusteredVariantsFromMongoStep)
-                                .next(processRSMergeCandidatesStep)
-                                .next(processRSSplitCandidatesStep)
-                                .next(clearRSMergeAndSplitCandidatesStep)
-                                .next(clusteringNonClusteredVariantsFromMongoStep)
-                                .next(backPropagateRSStep)
-                                .build();
+                .incrementer(new RunIdIncrementer())
+                //We need the dummy step here because Spring won't conditionally start the first step
+                .start(dummyStep)
+                .next(jobExecutionDecider)
+                    .on("TRUE")
+                    .to(new FlowBuilder<SimpleFlow>("remappedAssemblyClusteringFlow")
+                            .start(clusteringClusteredVariantsFromMongoStep)
+                            .next(processRSMergeCandidatesStep)
+                            .next(processRSSplitCandidatesStep)
+                            .next(clearRSMergeAndSplitCandidatesStep)
+                            .next(clusteringNonClusteredVariantsFromMongoStep)
+                            .next(backPropagateNewRSStep)
+                            .next(backPropagateSplitMergedRSStep).build())
+                    .on("*").end()
```

Comment:
Yeah, that's something that I tried first but it didn't work and it took me hours to debug :frowning_face:  That is because Spring job definition wants you to spell out the entire flow for both true and false conditions.

Reply:
I figured Spring was forcing your hand in some way... thanks for explaining!